File: 1.1/src/cz/mg/vulkan/PFNvkCmdClearAttachments.java
package cz.mg.vulkan;

public class PFNvkCmdClearAttachments extends VkFunctionPointer {
    public PFNvkCmdClearAttachments() {
    }

    protected PFNvkCmdClearAttachments(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdClearAttachments(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdClearAttachments(long value) {
        setValue(value);
    }

    public PFNvkCmdClearAttachments(VkInstance instance) {
        super(instance, new VkString("vkCmdClearAttachments"));
    }

    public void call(VkCommandBuffer commandBuffer, VkUInt32 attachmentCount, VkClearAttachment pAttachments, VkUInt32 rectCount, VkClearRect pRects){
        call(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), attachmentCount != null ? attachmentCount.getVkAddress() : VkPointer.getNullAddressNative(), pAttachments != null ? pAttachments.getVkAddress() : VkPointer.NULL, rectCount != null ? rectCount.getVkAddress() : VkPointer.getNullAddressNative(), pRects != null ? pRects.getVkAddress() : VkPointer.NULL);
    }

    protected static native void call(long vkaddress, long commandBuffer, long attachmentCount, long pAttachments, long rectCount, long pRects);
}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkCmdClearAttachments.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdClearAttachments_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong attachmentCount, jlong pAttachments, jlong rectCount, jlong pRects){
    (void)env;
    (void)jc;
    PFN_vkCmdClearAttachments f = (PFN_vkCmdClearAttachments)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((uint32_t*)jniLongToPointer(attachmentCount)),
        ((VkClearAttachment*)jniLongToPointer(pAttachments)),
        *((uint32_t*)jniLongToPointer(rectCount)),
        ((VkClearRect*)jniLongToPointer(pRects))
    );
}


void Java_cz_mg_vulkan_PFNvkCmdClearAttachments_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jint attachmentCount, jlong pAttachments, jint rectCount, jlong pRects){
    (void)env;
    (void)jc;
    PFN_vkCmdClearAttachments f = (PFN_vkCmdClearAttachments)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        (uint32_t)attachmentCount,
        ((VkClearAttachment*)jniLongToPointer(pAttachments)),
        (uint32_t)rectCount,
        ((VkClearRect*)jniLongToPointer(pRects))
    );
}

